js/main.js

```diff
@@ -23,38 +29,65 @@ function handleSubmit(event) {
     notes,
   };
 
-  formValues.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(formValues);
-
-  $photo.setAttribute('src', 'images/placeholder-image-square.jpg');
-
-  $form.reset();
-
   $ul.prepend(renderEntry('entries'));
   viewSwap($entryView);
 
   toggleNoEntries();
+
+  if (data.editing === null) {
+    renderEntry('entries');
+    formValues.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(formValues);
+    console.log(formValues.entryId);
+  } else if (data.editing !== null) {
+    formValues.entryId = data.editing.entryId;
+
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === formValues.entryId) {
+        data.entries[i] = formValues;
+      }
+    }
+
+    const newDOM = renderEntry(formValues);
+    const existingLi = document.querySelector(
+      `li[data-entry-id=" data.editing"]`
+    );
+
+    if (existingLi) {
+      existingLi.replaceWith(newDOM);
+      const $updateEntry = document.getElementById('entry-form');
+      $updateEntry.innerText = 'New Entry';
```

Comment:
I don't think this is the correct element? Your `div` with the `data-view="entry-form"` is getting selected by this. You want your `h2` that says "New entry" to be the one that gets targeted by this